### PR TITLE
[CI] update image diff framework and update to node 20

### PR DESCRIPTION
**Pull request by RaananW**
This moves us away from jest-screenshot (that is using native-image-diff) to a different framework (using pixelmatch)

There are difference with the way the old framework deals with transparency and the way the new one does. Some other images had to be updated as well.

This PR also includes changes to the project's dependencies  to allow running the repo in a node 20 environment.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13944/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13944/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13944/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13944/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13944/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13944/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13944/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13944/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13944/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13944/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

File: packages/tools/tests/test/visualization/ReferenceImages/prepass-ssao-shadow-only.png
Comment (sebavan) [inline packages/tools/tests/test/visualization/ReferenceImages/prepass-ssao-shadow-only.png:1]: Not sure why we are losing alpha ?

**Review by sebavan — COMMENTED** [review]
LGTM aside a couple image diffs

**Review by sebavan — COMMENTED** [review]
LGTM aside a couple image diffs

File: packages/tools/tests/test/visualization/ReferenceImages/shadowOnlyMaterial.png
[the remark this answers is not in the document]
Comment (RaananW) [reply packages/tools/tests/test/visualization/ReferenceImages/shadowOnlyMaterial.png:1]: discussed offline

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Review by sebavan — APPROVED** [review]

File: jest.visualization.setup.afterEnv.ts [hunk 1]
```diff
@@ -1,3 +1,11 @@
-import { setupJestScreenshot } from "jest-screenshot";
-setupJestScreenshot();
+// import { setupJestScreenshot } from "jest-screenshot";
+// setupJestScreenshot();
```
Comment (bghgary) [inline jest.visualization.setup.afterEnv.ts:2]: Remove commented out code?

Comment (RaananW) [reply jest.visualization.setup.afterEnv.ts]: Done :-)

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

File: packages/tools/tests/test/visualization/visualization.utils.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as jestScreenshot from "jest-screenshot";
+// import * as jestScreenshot from "jest-screenshot";
```
Comment (bghgary) [inline packages/tools/tests/test/visualization/visualization.utils.ts:1]: This one too?

Comment (RaananW) [reply packages/tools/tests/test/visualization/visualization.utils.ts]: 👌

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Comment by bjsplat** [comment]
Reviewer - this PR has made changes to one or more package.json files.

**Pull request merged**